src/Features/CSharp/Portable/Completion/CompletionProviders/CrefCompletionProvider.cs

```diff
@@ -312,8 +310,37 @@ private static TextSpan GetCompletionItemSpan(SourceText text, int position)
                 rules: GetRules(insertionText));
         }
 
-        protected override Task<CompletionDescription> GetDescriptionWorkerAsync(Document document, CompletionItem item, CancellationToken cancellationToken)
-            => SymbolCompletionItem.GetDescriptionAsync(item, document, cancellationToken);
+        protected override async Task<CompletionDescription> GetDescriptionWorkerAsync(Document document, CompletionItem item, CancellationToken cancellationToken)
+        {
+            var position = SymbolCompletionItem.GetContextPosition(item);
+            var tree = await document.GetSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
+            if (!tree.IsEntirelyWithinCrefSyntax(position, cancellationToken))
+            {
+                return null;
+            }
+
+            var token = tree.FindTokenOnLeftOfPosition(position, cancellationToken, includeDocumentationComments: true)
+                            .GetPreviousTokenIfTouchingWord(position);
+
+            // To get a Speculative SemanticModel (which is much faster), we need to 
+            // walk up to the node the DocumentationTrivia is attached to.
+            var parentNode = token.Parent.FirstAncestorOrSelf<DocumentationCommentTriviaSyntax>()?.ParentTrivia.Token.Parent;
+            _testSpeculativeNodeCallbackOpt?.Invoke(parentNode);
```

Comment:
This entire method seems hugely complex.  Can we not store more information in the item to make it so that recovery at this point is super simple to do?